### PR TITLE
Added the possibility to create directories

### PR DESCRIPTION
**Pull request by rvmourik**

### CODE REVIEW

File: src/RackspaceAdapter.php [hunk 1]
```diff
@@ -162,7 +162,23 @@ public function deleteDir($dirname)
...
+
+        if ($config->has('headers')) {
+            $headers = $config->get('headers');
+        }
+
+        if (!key_exists('Contents', $headers)) {
+            $headers['Contents'] = 'None';
+        }
+
+        if (!key_exists('Content-Type', $headers)) {
```
Comment (frankdejonge) [inline src/RackspaceAdapter.php:167]: Shouldn't this always be `application/directory`?

File: src/RackspaceAdapter.php [hunk 2]
```diff
@@ -162,7 +162,23 @@ public function deleteDir($dirname)
...
     {
-        return ['path' => $dirname];
+        $headers = [];
+
+        if ($config->has('headers')) {
+            $headers = $config->get('headers');
+        }
+
+        if (!key_exists('Contents', $headers)) {
+            $headers['Contents'] = 'None';
```
Comment (frankdejonge) [inline src/RackspaceAdapter.php]: Shouldn't this always be `None`?

**Comment by frankdejonge** [comment]
Travis didn't run yet, however I have the feeling it's not going to pass since the tests have not been corrected to account for this case.

File: src/RackspaceAdapter.php (hunk 2 above)
Comment (rvmourik) [inline src/RackspaceAdapter.php]: I wanted it to be dynamic, let's say the headers are already send through the config (for some reason). Maybe a implementation based on RackSpace but with slidely different values, one would be able to pass this through the config without being worried it's overwritten with 'None', the same goes for 'application/directory' in the Content-Type.

**Comment by rvmourik** [comment]
Okay, i'm not very familiar with Travis and Unit testing (yet), so is this something that needs work from my end or yours?

File: src/RackspaceAdapter.php (hunk 2 above)
Comment (frankdejonge) [inline src/RackspaceAdapter.php]: Alternatively, shouldn't it just have no content at all? Or is this in the docs as well?

**Comment by frankdejonge** [comment]
@rvmourik the test case needs to be updated which is located here: https://github.com/thephpleague/flysystem-rackspace/blob/master/tests/RackspaceAdapterTests.php#L158

**Comment by rvmourik** [comment]
@frankdejonge I removed the Contents None (our Cloud support department gave this as something that needed to be present, but just tested it, and it works without). I also changed the test to an assertTrue, but it still fails.

File: src/RackspaceAdapter.php [hunk 3]
```diff
@@ -162,7 +162,19 @@ public function deleteDir($dirname)
      */
     public function createDir($dirname, Config $config)
     {
-        return ['path' => $dirname];
+        $headers = [];
+
+        if ($config->has('headers')) {
+            $headers = $config->get('headers');
+        }
```
Comment (frankdejonge) [inline src/RackspaceAdapter.php]: this can be changed to:

``` php
$headers = $config->get('headers', []);
```

Comment (rvmourik) [inline src/RackspaceAdapter.php]: done

**Comment by rvmourik** [comment]
@frankdejonge can you tell me what needs to be done about the test in order to make it pass? Thanks.

**Comment by frankdejonge** [comment]
@rvmourik I've merged and corrected the PR. Thanks!

**Comment by rvmourik** [comment]
@frankdejonge thanks!

**Pull request merged**